api/vm/query.ts

```diff
@@ -17,8 +17,9 @@ export function compile(query: string, projectDir?: string): string {
       compiler: (code, path) => compilers.compile(code, path)
     });
     const indexScript = genIndex(
```

Comment:
This isn't behaving quite the same as the vm/compile script - it probably doesn't matter that much as we don't really use this at all, but here the genIndex from the same version as API will be used. Maybe a comment for now will suffice as I think we need to rethink this whole query compilation piece.

Reply:
good point! leaving a comment for now, I do agree query compilation needs rethinking.